sql: include sampled stats in TestSampledStatsCollection

asubiotto opened the pull request:
Depends on #59992, which is required for this new regression test to pass.

TestSampledStatsCollection would previously only check that stats that are
collected regardless of the sample rate are returned. These types of stats
(rows/bytes read) are propagated using metadata, rather than the trace.

This resulted in us silently failing to collect any stats when sampling was
enabled once the tracing mode was reverted back to legacy. To avoid this kind
of thing happening again, this commit adds a check that max memory usage is
reported to be non-zero.

Release note: None (this is a new feature that has no user impact yet)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60424)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
The last commit <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60424#-:-MTBcybcE9uYE9ld2f6-:b-xegzy9)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/instrumentation_test.go, line 83 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/60424#-MTBcj9M1bLLR64g4Nao:-MTBcj9M1bLLR64g4Nap:bwvcfas) ([raw file](https://github.com/cockroachdb/cockroach/blob/dcd49506f32954b2f45567458d42935b6f58d49d/pkg/sql/instrumentation_test.go#L83)):*
> ```Go
> 		require.Equal(t, int64(1), stats.mu.data.ExecStatCollectionCount, "expected to have collected exactly one set of execution stats")
> 		require.Greater(t, stats.mu.data.RowsRead.Mean, float64(0), "expected statement to have read at least one row")
> 		require.Greater(t, stats.mu.data.MaxMemUsage.Mean, float64(0), "expected statement to have requested bytes")
> ```

nit: to me "to have requested bytes" reads like it is more about BytesRead and not MaxMemUsage, maybe `expected statement to request some RAM` or something like that?


<!-- Sent from Reviewable.io -->

Commit:
sql: include sampled stats in TestSampledStatsCollection

TestSampledStatsCollection would previously only check that stats that are
collected regardless of the sample rate are returned. These types of stats
(rows/bytes read) are propagated using metadata, rather than the trace.

This resulted in us silently failing to collect any stats when sampling was
enabled once the tracing mode was reverted back to legacy. To avoid this kind
of thing happening again, this commit adds a check that max memory usage is
reported to be non-zero.

Release note: None (this is a new feature that has no user impact yet)

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60424)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/sql/instrumentation_test.go, line 83 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/60424#-MTBcj9M1bLLR64g4Nao:-MTp_TUB8-0sVxagUnA4:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dcd49506f32954b2f45567458d42935b6f58d49d/pkg/sql/instrumentation_test.go#L83)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: to me "to have requested bytes" reads like it is more about BytesRead and not MaxMemUsage, maybe `expected statement to request some RAM` or something like that?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

asubiotto commented:
TFTR

bors r=yuzefovich
